README.md

```diff
@@ -1,18 +1,1481 @@
+
+## Using the Okta CDN
+
+Loading our assets directly from the CDN is a good choice if you want an easy way to get started with okta-auth-js, and don't already have an existing build process that leverages [npm](https://www.npmjs.com/) for external dependencies.
+
+To use the CDN, include links to the JS and CSS files in your HTML:
+
+```html
+<!-- Latest CDN production Javascript: 1.6.0 -->
+<script
+  src="https://ok1static.oktacdn.com/assets/js/sdk/okta-auth-js/1.6.0/okta-auth-js.min.js"
+  type="text/javascript"></script>
+```
+
+The `okta-auth-js.min.js` file will expose a global `OktaAuth` object which can be used to bootstrap the client:
+
+```javascript
+var authClient = new OktaAuth({/* configOptions */});
+```
+
+## Using the npm module
+
+Using our npm module is a good choice if:
+- You have a build system in place where you manage dependencies with npm
+- You do not want to load scripts directly from third party sites
+
+To install [@okta/okta-auth-js](https://www.npmjs.com/package/@okta/okta-auth-js):
+
+```bash
+# Run this command in your project root folder.
+[project-root-folder]$ npm install @okta/okta-auth-js --save
+```
+
+After running `npm install`:
+
+The minified auth client will be installed to `node_modules/@okta/okta-auth-js/dist`. You can copy the `dist` contents to a publicly hosted directory. However, if you're using a bundler like [Webpack](https://webpack.github.io/) or [Browserify](http://browserify.org/), you can simply import the module using CommonJS.
+
+```javascript
+var OktaAuth = require('@okta/okta-auth-js');
+var authClient = new OktaAuth(/* configOptions */);
+```
+
+# API
+
+## new OktaAuth(config)
+
+Creates a new instance of the Okta Auth Client with the provided options. The client has many [config options](#configuration). The only required option to get started is `url`, the base url for your Okta domain.
+
+  - `config` - Options that are used to configure the client
+
+```javascript
+var authClient = new OktaAuth({url: 'https://acme.okta.com'});
+```
+
+## signIn(options)
+
+The goal of an authentication flow is to set an Okta session cookie on the user's browser. The flow is started using `signIn`.
+
+  - `username` - User’s non-qualified short-name (e.g. dade.murphy) or unique fully-qualified login (e.g dade.murphy@example.com)
+  - `password` - The password of the user
+
+```javascript
+authClient.signIn({
+  username: 'some-username',
+  password: 'some-password'
+})
+.then(function(transaction) {
+  if (transaction.status === 'SUCCESS') {
+    authClient.session.setCookieAndRedirect(transaction.sessionToken); // Sets a cookie on redirect
+  } else {
+    throw 'We cannot handle the ' + transaction.status + ' status';
+  }
+})
+.fail(function(err) {
+  console.error(err);
+});
+```
+
+## signOut()
+
+This is an alias for `session.close`, which ends the current session.
```

Comment:
Can we just remove this one (or the other one) - I like the idea of just choosing one and not exposing the other.